js/main.js

```diff
@@ -80,10 +104,25 @@ function openPreviousView(event) {
   }
 }
 
+function handleEdit(event) {
+  if (event.target.tagName === 'I') {
+    openEntryForm();
+    $formTitle.textContent = 'Edit Entry';
+    var entry = event.target.closest('.entry');
+    var entryID = entry.getAttribute('data-entry-id');
+    var dataEntry = data.entries.length - entryID;
```

Comment:
You should not depend on an `id` to be the same as its index like this. For instance, what if eventually the item with the id `4` gets deleted? Now this won't work. Instead, you should loop over your list of entries and find the entry with the matching id.

Reply:
Changes made! I was also using the same tactic on lines 22~25 to replace the previous entry with the updated entry on the entry list. Changed that to a for loop as well.